src/FormErrorInterface.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Form;
+
+/**
+ * FormErrors represents a form errors collection.
+ */
+interface FormErrorInterface
+{
+    /**
+     * Add error for the specified attribute.
+     *
+     * @param string $attribute attribute name.
+     * @param string $error attribute error message.
+     */
+    public function addError(string $attribute, string $error): void;
+
+    /**
+     * Removes errors for all attributes.
+     */
+    public function clear(): void;
+
+    /**
+     * Returns the errors for single attribute.
+     *
+     * @param string $attribute attribute name. Use null to retrieve errors for all attributes.
+     *
+     * @return array
+     */
+    public function getError(string $attribute): array;
+
+    /**
+     * Returns the errors for all attributes.
+     *
+     * @return array errors for all attributes or the specified attribute. null is returned if no error.
```

Comment:
I'd suggest returning empty array for no errors instead of using `null`.

```suggestion
     * @return array Errors for all attributes.
```

Reply:
Done.